src/components/TodoItem.tsx

```diff
@@ -0,0 +1,122 @@
+import classNames from 'classnames';
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo,
+  isProcessing?: boolean,
+  onDelete: (todoId: number) => void,
+  onRename: (todoId: number, title: string) => void,
+  onComplete: (todo: Todo) => void,
+};
+
+export const TodoItem: React.FC<Props> = React.memo(
+  ({
+    todo,
+    isProcessing,
+    onDelete,
+    onRename,
+    onComplete,
+  }) => {
+    const [isRenaming, setIsRenaming] = useState(false);
+    const [title, setTitle] = useState(todo.title);
+    const [newTitle, setNewTitle] = useState(todo.title);
+    const renameInput = useRef<HTMLInputElement>(null);
+
+    const onRenameHander = async (event: React.FormEvent<HTMLFormElement>) => {
```

Comment:
```suggestion
    const handleRename = async (event: React.FormEvent<HTMLFormElement>) => {
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364